services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProviderResource.java

```diff
@@ -130,17 +131,21 @@ public Response delete() {
 
     @PUT
     @Consumes(MediaType.APPLICATION_JSON)
-    public Response update(IdentityProviderRepresentation providerRep) {
+    public Response update(IdentityProviderRepresentation rep) {
         IdentityProviderModel identityProvider = getIdentityProviderModel();
 
-        Response response = getIdentityProviderResource(identityProvider).update(providerRep);
+        if (!rep.getAlias().equals(identityProvider.getAlias())) {
+            throw ErrorResponse.error("Identity provider not assigned to the organization.", Status.NOT_FOUND);
+        }
```

Comment:
Yeah, but I think we shouldn't for organizations. It is actually very confusing how we manage identity providers. There are also expectations from the `IdentityProviderResource` that the `internalId` can be different and for now, keeping the alias unchanged, works better I would say.

Reply:
Should we then add the check that `internalId` is null for the provided `rep` or it matches the `idp` linked to the org?

Reply:
TBH, I have no clue because this is kinda confusing in the `IdentityProviderResource`.

I'm more thinking of enforcing here that you can't change the alias we always use the alias to lookup/bind the broker for an org.

If that works for you, let's keep that way and perhaps review this before 26.

Reply:
That works, I've created https://github.com/keycloak/keycloak/issues/28833 to track it and I'll send PR once this one is merged.